contracts/cryptoblades.sol

```diff
@@ -519,10 +519,11 @@ contract CryptoBlades is Initializable, AccessControlUpgradeable {
         // they meditate on what they've learned
         for(uint256 i = 0; i < characters.balanceOf(msg.sender); i++) {
             uint256 char = characters.tokenOfOwnerByIndex(msg.sender, i);
-            if(xpRewards[char] < 65536)
-                characters.gainXp(char, uint16(xpRewards[char]));
-            else
-                characters.gainXp(char, 65535);
+            uint256 xpRewardsToClaim = xpRewards[char];
+            if (xpRewardsToClaim > 65535) {
+                xpRewardsToClaim = 65535;
+            }
+            characters.gainXp(char, uint16(xpRewardsToClaim));
```

Comment:
As I detailed in #74, it is unlikely to be an issue, but it is also good practice in general just in case when it might be an issue.

By nature of smart contract code and the blockchain, the concept of an atomic swap does not exist and is not needed anyway. It is simply enough to move `xpRewards[char] = 0` to just after `uint256 xpRewardsToClaim = xpRewards[char]`. See my suggested changes.